### PR TITLE
[MVT] Go global

### PR DESCRIPTION
**Pull request by procrastinatio**
[Test link with global WMS/WMTS and Swiss Extent](https://mf-chsdi3.int.bgdi.ch/mvt/shorten/7eaf3f8386)

1. Check if you can add any WMS around the world
2.  Check if you can still use LV03/LV95 to go home.

Should fixe https://github.com/geoadmin/mf-geoadmin3/issues/4631

<jenkins>[Test link](https://mf-geoadmin3.int.bgdi.ch/mvt_global_extent/index.html)</jenkins>

### CODE REVIEW

**Comment by oterral** [comment]
@procrastinatio  doesn't it break the swiss coordinate detection ?

**Comment by oterral** [comment]
I will merge if @davidoesch is ok with that.

**Comment by davidoesch** [comment]
> @oterral Jein, if coordinates are metrics (>180), it looks if inside the LV03 and LV95 bboxes, and if not, it is assumed it is epsg:3857. But any LV03 or LV95 coordinates are also valid epsg:3857 coordinates.

can we fix this with https://github.com/geoadmin/mf-geoadmin3/issues/4618#issuecomment-444437915 proposal?

**Comment by davidoesch** [comment]
> I will merge if @davidoesch is ok with that.

I wonder which sideeffect it will have. anyway, the application has to start centered on switzerland as MVT branch currently does....

**Comment by loicgasser** [comment]
This will break coords detection. We do need a an extent for Switzerland.

**Comment by loicgasser** [comment]
It can be another variable though. But you need to adapt other part of the code. Notably the search service, search types also I believe and maybe some more.

**Comment by davidoesch** [comment]
if this PR (which was about supporting about external WMS/WMTS) breaks most cases of https://help.geo.admin.ch/?id=24 

-> a no go. we need a solution here which works with LV95 without additional user interaction

**Comment by procrastinatio** [comment]
@loicgasser So, search is the only issue? So this is not a problem. See my two last comments on https://github.com/geoadmin/mf-geoadmin3/issues/4618#issuecomment-444437915

**Comment by loicgasser** [comment]
the problem is here:
https://github.com/geoadmin/mf-geoadmin3/blob/mvt/src/components/search/SearchService.js#L186

everything will always be interpreted as `EPSG:2056` since the point is always contained within the extent it will NEVER be matched against `EPSG:21781`.

Just try to input 700499.35, 182750.43
This should be interpreted as `EPSG:21781` it works on mvt but not on this branch.
So keep a ref to the swiss extent in another variable and replace in the SerachService.

Please have a look at what has been done in https://github.com/geoadmin/mf-geoadmin3/pull/4613/files

And I agree she don't care about inputting WGS coords.

so keep global extent to test lon/lat, but use swiss extent to test swiss coords.

**Comment by procrastinatio** [comment]
@loicgasser Also my PR https://github.com/geoadmin/mf-geoadmin3/pull/4625 and my comment in https://github.com/geoadmin/mf-geoadmin3/issues/4618
I we assume the LV03 and LV95 bboxes as swiss, we cannot access the two area of the Gulf of Guniea and South Sudan with Webmercator. 

OK, I'll add a varaible swiss_extent to the mix.

**Comment by procrastinatio** [comment]
And what do we do with Skyguide black magick? We were assuming something like 7 47 to be interpreted as 7°E and 47°N, but it could be anything else.

**Comment by davidoesch** [comment]
> And what do we do with Skyguide black magick? We were assuming something like 7 47 to be interpreted as 7°E and 47°N, but it could be anything else.

We support officiall what is in
https://help.geo.admin.ch/?id=24&lang=de

46.97984 6.60757 (DD) ◈ 46°58.7904' 6°36.4542' (DM) ◈ 46°58'47.424'' 6°36'27.252'' (DMS, double single quote) ◈ 46°58'47.424" 6°36'27.252" (DMS, double quote)

so the y need to enter 7.0 47. 0

File: Makefile [hunk 1]
```diff
@@ -111,26 +111,19 @@ TRANSLATE_EMPTY_JSON ?= src/locales/empty.json
...
 DEFAULT_ELEVATION_MODEL ?= COMB
 
-# Map (Mercator)
+# Map variables (Mercator)
 DEFAULT_EPSG = EPSG:3857
 DEFAULT_EPSG_EXTEND = '[-20037508.3428, -20037508.3428, 20037508.3428, 20037508.3428]'
-DEFAULT_EXTENT = '[558147.7958306982, 5677741.814085617, 1277662.36597472, 6152731.529704217]'
+# Only for WMS services
+# The Swiss bounds expressed as $DEFAULT_EPSG 
+SWISS_EXTENT = '[558147.7958306982, 5677741.814085617, 1277662.36597472, 6152731.529704217]'
```
Comment (loicgasser) [inline Makefile]: but then use it no?

**Comment by loicgasser** [comment]
@davidoesch this is now for the MVT branch. But it still requires some changes and data is incorrect at global scale.

**Comment by davidoesch** [comment]
yes at MVT I still not global coverage: https://mf-chsdi3.int.bgdi.ch/mvt/shorten/7ec2e0dfcf 

image


or did I understand sth wrong?

**Comment by procrastinatio** [comment]
Need more work.

- All tests in the branch `mvt` are still EPSG:2056 (see https://github.com/geoadmin/mf-geoadmin3/blob/mvt/test/specs/Loader.spec.js#L63)
- Too many black magick to guess the coordinates which need to be removed
- We may search outside `swissExtent`, do we?

**Comment by davidoesch** [comment]
Wenn may ... But for test viewer stick to simplicity

**Comment by oterral** [comment]
@procrastinatio outside swissextent, you use lon lat coord

**Comment by procrastinatio** [comment]
Test
[Test with many WMS/WMTS](https://mf-chsdi3.int.bgdi.ch/mvt/shorten/7eceae85c1)

`37UDB 13433 79521`   Red Square, Moscow
`40.68921, -74.04454`  or  `40°41′21.139″N 74°02′40.332″W`  Liberty Statue, New York

`6.96948 46.9712`  Not in  lat/long, but assumed in Switzerland

`6.96948W 46.9712N`If you want to go west of Brest, use quadrant

**Comment by procrastinatio** [comment]
jenkins test

**Comment by procrastinatio** [comment]
@loicgasser `extent`defines where coordinates are valid. `swissExtent`defines only the area where `LV03`and `LV95` are valid, and where you may use `lat/long`and `lon/lat` without quadrant indication (normally `7.5° 46.7°`should be interpreted as `7.5°N 46.7°W` but we are nice to the user)

**Comment by oterral** [comment]
@loicgasser are we good with this PR?  the seacrh of coordinates david mentionned works

**Comment by oterral** [comment]
@davidoesch are you agree with this functionnality?

**Comment by davidoesch** [comment]
@oterral 
-> OK 
I tested: https://mf-geoadmin3.int.bgdi.ch/mvt_global_extent/index.html?

and they work four our official supported coordinate imput like defined under https://help.geo.admin.ch/?id=24 

CH1903+ / LV95: 2600'000 1200'000 ◈ 2600000 1200000 ◈ 2600 000 1200 000 ◈ 2600000/1200000 ◈ 2600000 / 1200000 ◈ 2600000,1200000 ◈ 2600000 1200000 ◈ 1200000 2600000

CH1903 / LV03: 600'000 200'000 ◈ 600000 200000 ◈ 600 000 200 000 ◈ 600000/200000 ◈ 600000 / 200000 ◈ 600000,200000 ◈ 600000 200000 ◈ 200000 600000

WGS 84 (lat/lon): 46.97984 6.60757 (DD) ◈ 46°58.7904' 6°36.4542' (DM) ◈ 46°58'47.424'' 6°36'27.252'' (DMS, double single quote) ◈ 46°58'47.424" 6°36'27.252" (DMS, double quote)

MGRS: "32TLT 98757 23913"

what3words: "zufall.anders.blaumeise"


and we do now support global WGS84 which makes sense


-> LGTM

**Pull request merged**